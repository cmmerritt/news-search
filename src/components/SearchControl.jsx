/* eslint-disable max-len */
import React from 'react';
import PropTypes from 'prop-types';

const SearchControl = ({ text, onChange, onSubmit }) => {
  return <form>
    <label htmlFor="text">Search by keyword: </label>
    <input id="text" placeholder="Search" type="text" name="text" value={text} onChange={onChange} onSubmit={onSubmit}/>
  </form>;
};

SearchControl.propTypes = {
  text: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired
};

export default SearchControl;
