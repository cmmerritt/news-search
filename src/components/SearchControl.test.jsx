/**
 * @jest-environment jsdom
 */

import React from 'react';
import { render } from '@testing-library/react';
import SearchControl from './SearchControl';

describe('SearchControl component', () => {
  it('renders SearchControl', () => {
    const { asFragment } = render(
      <SearchControl text="dinosaur"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
