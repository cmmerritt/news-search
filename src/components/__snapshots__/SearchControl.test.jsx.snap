// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchControl component renders SearchControl 1`] = `
<DocumentFragment>
  <form>
    <label
      for="text"
    >
      Search by keyword: 
    </label>
    <input
      data-testid="textInput"
      id="text"
      name="text"
      placeholder="Search"
      type="text"
      value="dinosaur"
    />
  </form>
</DocumentFragment>
`;
